timbuctoo-instancev4/src/main/java/nl/knaw/huygens/timbuctoo/v5/dropwizard/endpoints/JsonLdEditEndpoint.java

```diff
@@ -0,0 +1,118 @@
+package nl.knaw.huygens.timbuctoo.v5.dropwizard.endpoints;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.jsonldjava.core.JsonLdError;
+import com.github.jsonldjava.core.JsonLdOptions;
+import com.github.jsonldjava.core.JsonLdProcessor;
+import com.google.common.base.Charsets;
+import nl.knaw.huygens.timbuctoo.v5.bdbdatafetchers.dto.CursorQuad;
+import nl.knaw.huygens.timbuctoo.v5.dataset.DataSetFactory;
+import nl.knaw.huygens.timbuctoo.v5.dataset.Direction;
+import nl.knaw.huygens.timbuctoo.v5.dataset.ImportManager;
+import nl.knaw.huygens.timbuctoo.v5.dataset.QuadStore;
+import nl.knaw.huygens.timbuctoo.v5.datastores.exceptions.DataStoreCreationException;
+import nl.knaw.huygens.timbuctoo.v5.filestorage.exceptions.LogStorageFailedException;
+import nl.knaw.huygens.timbuctoo.v5.jsonldimport.Entity;
+import nl.knaw.huygens.timbuctoo.v5.jsonldimport.GenerateRdfPatchFromJsonLdEntity;
+import nl.knaw.huygens.timbuctoo.v5.util.RdfConstants;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.MediaType;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Optional;
+
+@Path("/v5/{user}/{dataset}/upload/jsonld")
+public class JsonLdEditEndpoint {
+
+  private DataSetFactory dataSetFactory;
+  private ObjectMapper objectMapper;
+
+  private String userId;
+
+  private String dataSet;
+
+  public JsonLdEditEndpoint(DataSetFactory dataSetFactory, ObjectMapper objectMapper) {
+    this.dataSetFactory = dataSetFactory;
+
+    this.objectMapper = objectMapper;
+  }
+
+  @PUT
+  public void submitChanges(String jsonLdImport,
+                            @PathParam("user") String userId,
+                            @PathParam("dataset") String dataSet)
+    throws DataStoreCreationException, LogStorageFailedException, IOException, JSONException, JsonLdError {
+
+    if (!integrityCheck(jsonLdImport)) {
+      throw new JSONException("JSON-LD failed integrity check");
```

Comment:
This will return a 500 (i.e. "the server screwed up") without any information for the client. It's more clear to the client to return `Response.status(400).entity("")...`